### PR TITLE
CI/DOC: Building documentation with azure

### PR DESCRIPTION
**Pull request by datapythonista**
- [X] xref #22766, #26574
- [ ] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by datapythonista** [comment]
This should be ready now. This will build the docs with azure for all CI builds (PRs and master), and will publish them as GitHub pages for master (not for PRs).

I created a repo https://github.com/pandas-dev/pandas-dev.github.io . I made it private, to avoid people getting confused and opening issues or opening PRs for the docs there (it happened in the past with the sphinx theme repo).

The master docs will first be pushed once this is merged. The url to see the docs is: https://pandas-dev.github.io/. The url can easily be changed by a custom domain at: https://github.com/pandas-dev/pandas-dev.github.io/settings (my preference would be to use dev.pandas.io).

In a follow up, we can make sure the docs build on windows too if we want (see #26574). We should just move the new job to `ci/azure/docs.yml` and call it twice for Linux and Windows. We'll only need to add an extra condition to the publishing of docs, so it's just pushed from Linux.

CC: @pandas-dev/pandas-core @stijnvanhoey

**Comment by datapythonista** [comment]
@vtbassmatt implemented pandas docs build following your blog post, if you've got any feedback, please let us know.

**Review by TomAugspurger — COMMENTED** [review]
Thanks @datapythonista, This looks good.

File: azure-pipelines.yml [hunk 1]
```diff
@@ -116,3 +116,63 @@ jobs:
...
     condition: true
+
+- job: 'Docs'
+  pool:
+    vmImage: ubuntu-16.04
+  timeoutInMinutes: 90
+  steps:
+  - script: |
+      echo '##vso[task.setvariable variable=CONDA_ENV]pandas-dev'
+      echo '##vso[task.setvariable variable=ENV_FILE]environment.yml'
```
Comment (TomAugspurger) [inline azure-pipelines.yml]: What environment file is this?

Comment (datapythonista) [reply azure-pipelines.yml]: The one in the root, for dev environments (like locally). It's the one that contains the dependencies to build the docs.

Comment (TomAugspurger) [reply azure-pipelines.yml]: Oh of course. Should we re-use `ci/deps/travis-36-doc.yaml` for now (and rename that when the travis doc job is removed)?

Comment (datapythonista) [reply azure-pipelines.yml]: I thought travis was using `environment.yml`.

I'd personally use `environment.yml`, so we build in the CI with the same versions as locally, and remove `travis-36-doc.yaml` when we remove the travis job. Also it keep things simpler.

Any other advantage of using a specific dependencies file besides installing only the needed packages for the docs?

Comment (TomAugspurger) [reply azure-pipelines.yml]: > besides installing only the needed packages for the docs?

I think that's the primary benefit, but it's not a big deal to install extra deps (like test-only ones).

Stepping back, my main request is that we use the same env for building the travis & azure docs. I don't have a strong preference for whether that's `/environment.yaml` or `ci/deps/travis-36-doc.yaml`.

Comment (simonjayhawkins) [reply azure-pipelines.yml]: +1 for same on each

+1 for environment.yml

Comment (datapythonista) [reply azure-pipelines.yml]: That makes sense, I changed travis build to use `environment.yml` for the docs.

**Comment by jreback** [comment]
looks good @datapythonista thanks. note there are a bunch of warnings as well from recent changes, but othogonal.

Let's leave the original job around for a while to make sure this is completely working & updated.

Can change the contributing guide if you'd like now, eg. the path to the built dev-docs (or later ok too)

**Comment by datapythonista** [comment]
That make sense @jreback, we've got a sprint next week, I'll see if we can get rid of all the warnings in the docs and start validating that they're not reintroduced.

There is a problem with the CI of this change with the ssh key. That couldn't be tested in the PR, since the key is only available for master builds. I'll revert the pushing of the azure in a PR now, while we figure out what's wrong.

**Comment by jreback** [comment]
@datapythonista the build worked, but the push failed: https://dev.azure.com/pandas-dev/pandas/_build/results?buildId=12093

we might need to disable failing the builds until this is fixed

**Comment by jorisvandenbossche** [comment]
> I changed travis build to use environment.yml for the docs.

The doc build needs more that what is in that environment (we have additional optional dependencies that are used in the docs), so now a bunch of examples are failing

Please, if you change substantial things about the doc setup, can you at least check the travis / azure logs that things are actually working?

(I fully understand that that is tedious to do before merging each small doc fix, but this one was quite substantial ..)

**Pull request merged**